protocol/daemons/server/types/update_monitor.go

```diff
@@ -26,12 +27,18 @@ type UpdateMonitor struct {
 	disabled bool
 	// lock is used to synchronize access to the monitor.
 	lock sync.Mutex
+
+	// These fields are initialized in NewUpdateFrequencyMonitor and are not modified after initialization.
+	logger                   log.Logger
+	daemonStartupGracePeriod time.Duration
```

Comment:
I moved `daemonStartupGracePeriod` to be a member of the `UpdateMonitor` in order to write a unit test for the case of an unresponsive daemon.